### ISSUE TITLE
[BUG] [Pipeline] - streamline _build_train_config updates

### ISSUE THREAD

**Issue by Mars-Martiny** [Mars-Martiny/CITS5206_group1#94]
Options to loss functions added but logic leftover from previous implementation which did not give options was left in, cleaning up config function and also adding ability to pass extra arguments to the loss function (such as `gamma` value, for which there is a 'catch' in `get_loss_function` but no ability to actually pass a value to it when running the training loop).

**Comment by Mars-Martiny** [comment]
Fix accidentally pushed directly to dev but loss functionality cleaned up.
